### PR TITLE
Refactoring: Create config module

### PR DESCRIPTION
**Pull request by albertvillanova**
Refactorize configuration settings into their own module.

This could be seen as a Pythonic singleton-like approach. Eventually a config instance class might be created.

### CODE REVIEW

**Review by lhoestq — COMMENTED** [review]
Awesome thank you ! This is much cleaner

Also I'm pretty sure we don't need to check the os.environ twice for the cache directories. I think we can remove that

File: src/datasets/config.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+DEFAULT_HF_CACHE_HOME = os.path.join(XDG_CACHE_HOME, "huggingface")
+HF_CACHE_HOME = os.path.expanduser(os.getenv("HF_HOME", DEFAULT_HF_CACHE_HOME))
+DEFAULT_HF_DATASETS_CACHE = os.path.join(HF_CACHE_HOME, "datasets")
+# TODO: Why import in try and os.getenv twice in except?
+try:
+    from pathlib import Path
+
+    HF_DATASETS_CACHE = Path(os.getenv("HF_DATASETS_CACHE", DEFAULT_HF_DATASETS_CACHE))
+except (AttributeError, ImportError):
+    HF_DATASETS_CACHE = os.getenv(os.getenv("HF_DATASETS_CACHE", DEFAULT_HF_DATASETS_CACHE))
```
[the remark this answers is not in the document]
Comment (albertvillanova) [reply src/datasets/config.py:92]: Also putting the import from pathlib inside the try makes no sense, as this import is already used in many parts of the code without try...

[the remark this answers is not in the document]
Comment (lhoestq) [reply src/datasets/config.py:92]: yes totally

**Review by lhoestq — APPROVED** [review]
Thanks for the refactor :)

We can remove the double getenv, I'll open a PR

**Pull request merged**